docs/components/concepts/job-workers.md

```diff
@@ -77,6 +77,15 @@ When `Completing or failing jobs` with [variables](components/concepts/variables
 - When `Completing a job` the variables are propagated from the scope of the task to its higher scopes.
 - When `Failing a job` the variables are only created in the local scope of the task.
 
+:::tip Failing a job with variables
+There are several advantages when failing a job with variables. Consider the following use cases:
+
+- You can fail a job and raise an incident by setting the job `retries` to zero. In this case, it would be useful to
+  provide some additional details through a variable when the incident is analyzed.
+- You can partially complete a job and fail it. In this case, you can store store some information on the completed work
```

Comment:
❓ I wonder whether we should refer to `partially complete a job and fail it` differently because it could imply that partially completing a job is something the job worker should do, just like completing a job. But I don't know what wording would be more fitting.

Reply:
I agree here. Thoughts @koevskinikola? Happy to assist based on what you are trying to say here.

Reply:
That's a good point. Instead of:
> You can partially complete a job and fail it.
how about?
> You can fail a job if it can't be successfully completed.

It's a bit difficult to write something that doesn't insinuate that the job worker should be able to do partial work.

@christinaausley the idea is that in some use cases, the work can be split into more standalone parts. For example, the work may include a calculation that is then used for some other step that couldn't be done at that point in time. The user can then store this intermediate data in a variable and retrieve it when they retry the job. Of course, this "partitioning" of the job execution is completely up to how the user implements their job worker, so we shouldn't indicate in our documentation that we expect that in any way.

Reply:
How about something like:

```
If your job worker can split the job into smaller pieces and finish some but not all of these, then it can fail the job with variables to indicate which parts of the job were successfully finished and which parts weren't. Such a job should be failed with a positive number of retries so another job worker can pick it up again and continue where the other job worker left off. The job can be completed when all parts are finished by a job worker successfully.
```

Reply:
Nice @korthout!

Reply:
Was this thread resolved?

Reply:
I added this in my latest commit. Happy to go back and edit if you have additional comments @koevskinikola. Merging this in!